python/graphstorm/inference/ep_infer.py

```diff
@@ -136,20 +136,22 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
         if save_prediction_path is not None:
             if edge_id_mapping_file is not None:
                 g = loader.data.g
-                etype = infer_data.eval_etypes[0]
-                pred_shape = list(pred.shape)
-                pred_shape[0] = g.num_edges(etype)
-                pred_data = create_dist_tensor(pred_shape, dtype=pred.dtype,
-                                               name='predict-'+'-'.join(etype),
-                                               part_policy=g.get_edge_partition_policy(etype),
-                                               # TODO: this makes the tensor persistent in memory.
-                                               persistent=True)
-                # edges that have predictions may be just a subset of the
-                # entire edge set.
-                pred_data[loader.target_eidx[etype]] = pred.cpu()
-
-                pred = shuffle_predict(pred_data, edge_id_mapping_file, etype, get_rank(),
-                                       get_world_size(), device=device)
-            save_prediction_results(pred, save_prediction_path, get_rank())
+                shuffled_preds = {}
+                for etype, pred in preds.items():
+                    assert etype in infer_data.eval_etypes
+                    pred_shape = list(pred.shape)
+                    pred_shape[0] = g.num_edges(etype)
+                    pred_data = create_dist_tensor(pred_shape, dtype=pred.dtype,
+                        name='predict-'+'-'.join(etype),
+                        part_policy=g.get_edge_partition_policy(etype),
+                        # TODO: this makes the tensor persistent in memory.
+                        persistent=True)
```

Comment:
can we remove this? this will keep the distributed tensor persistent even if we will never use the tensor again.

Reply:
We can. But I am not sure whether DGL solved the bug of GC a DistTensor.

Reply:
I have changed it into False

Reply:
we should test it. if it still doesn't work, we need to inform the DGL team to fix it.